### PR TITLE
feat(resolveRef): new function

### PR DESCRIPTION
**Pull request by antfu**
This will enable some scenarios to have a function getter as `MaybeRef` (Refactor of the codebase will be done in future PRs).

For example, currently we have to:

```ts
const title = computed(() => `${foo} | template`)
useTitle(title)
```

Later it will be possible to have:

```ts
useTitle(() => `${foo} | template`)
```

### CODE REVIEW

File: packages/shared/resolveRef/index.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+category: Utilities
+related: resolveRef
```
Comment (webfansplz) [inline packages/shared/resolveRef/index.md:3]: related: resolveUnref

File: packages/shared/resolveUnref/index.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+category: Utilities
+related: resolveUnref
```
Comment (webfansplz) [inline packages/shared/resolveUnref/index.md:3]: related: resolveRef

File: packages/shared/resolveRef/index.md (hunk 1 above)
Comment (antfu) [inline packages/shared/resolveRef/index.md:3]:
```suggestion
related: resolveUnref
```

File: packages/shared/resolveUnref/index.md (hunk 2 above)
Comment (antfu) [inline packages/shared/resolveUnref/index.md:3]:
```suggestion
related: resolveRef
```

**Pull request merged**